impl/src/main/java/org/ehcache/impl/internal/spi/serialization/DefaultSerializationProvider.java

```diff
@@ -64,31 +64,25 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(DefaultSerializationProvider.class);
 
-  private final TransientProvider transientProvider;
-  private final PersistentProvider persistentProvider;
+  protected final Map<Class<?>, Class<? extends Serializer<?>>> serializers;
 
   final ConcurrentWeakIdentityHashMap<Serializer<?>, AtomicInteger> providedVsCount = new ConcurrentWeakIdentityHashMap<Serializer<?>, AtomicInteger>();
   final Set<Serializer<?>> instantiated = Collections.newSetFromMap(new ConcurrentWeakIdentityHashMap<Serializer<?>, Boolean>());
 
   public DefaultSerializationProvider(DefaultSerializationProviderConfiguration configuration) {
     if (configuration != null) {
-      transientProvider = new TransientProvider(configuration.getTransientSerializers());
-      persistentProvider = new PersistentProvider(configuration.getPersistentSerializers());
+      this.serializers = new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>(configuration.getDefaultSerializers());
     } else {
-      transientProvider = new TransientProvider(Collections.<Class<?>, Class<? extends Serializer<?>>>emptyMap());
-      persistentProvider = new PersistentProvider(Collections.<Class<?>, Class<? extends Serializer<?>>>emptyMap());
+      this.serializers = new LinkedHashMap<Class<?>, Class<? extends Serializer<?>>>(Collections.<Class<?>, Class<? extends Serializer<?>>>emptyMap());
     }
   }
 
   @Override
   public <T> Serializer<T> createKeySerializer(Class<T> clazz, ClassLoader classLoader, ServiceConfiguration<?>... configs) throws UnsupportedTypeException {
-    Serializer<T> serializer = (Serializer<T>)getUserProvidedSerializer(find(DefaultSerializerConfiguration.Type.KEY, configs));
+    Serializer<T> serializer = (Serializer<T>) getUserProvidedSerializer(find(DefaultSerializerConfiguration.Type.KEY, configs));
```

Comment:
Replacing this line with:
```java
DefaultSerializerConfiguration<T> configuration = find(DefaultSerializerConfiguration.Type.KEY, configs);
Serializer<T> serializer = getUserProvidedSerializer(configuration);
```
will resolve generics warnings.

Also applies to `createValueSerializer`